apps/src/code-studio/initApp/initApp.js

```diff
@@ -240,16 +240,24 @@ window.apps = {
     setInitialLevelSource: function (levelSource) {
       appOptions.level.lastAttempt = levelSource;
     },
+    // returns a Promise to the level source
     getLevelSource: function (currentLevelSource) {
-      var source;
-      if (window.Blockly) {
-        // If we're readOnly, source hasn't changed at all
-        source = Blockly.mainBlockSpace.isReadOnly() ? currentLevelSource :
-          Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
-      } else if (appOptions.getCode) {
-        source = appOptions.getCode();
-      }
-      return source;
+      return new Promise (function (resolve,reject) {
```

Comment:
Nit, remove space after `Promise` and add space between resolve/reject. Also, this could be an arrow function if you wanted.